Add trailing slashes to urls and redirect with names

bjrne opened the pull request:
Closes #157

bjrne committed:
Add trailing slashes to urls and redirect with names

Baschdl reviewed (changes requested):
You may also want to add a slash to `use_statistics/view`. The B button on the start page leads to a 404 but I think that the `href` is incorrect: `matching/signup_b/` (Done in PR for #147). logout with `matching/profile_staff/logout/` also doesn't work.

bjrne committed:
Merge branch 'staging' into 157-urls

bjrne committed:
Add trailing slash in statistics app

bjrne committed:
Address feedback

bjrne committed:
Merge branch 'staging' into 157-urls

bjrne commented:
Signup for B works for me. The logout url from the staff profile page is `/matching/logout/` for me and also works. Maybe there was a bug with `staff_profile` and `profile_staff`, but for me it works now.

bjrne committed:
Also add slashes to success urls

bjrne committed:
Remove unnecessary p_type from profile url

bjrne committed:
Fix typo

bjrne committed:
Use reverse_lazy because URLconf is not loaded yet when evaluating class

As per https://docs.djangoproject.com/en/3.1/ref/urlresolvers/#reverse-lazy

It is useful for when you need to use a URL reversal before your project’s URLConf is loaded. Some common cases where this function is necessary are:
- providing a reversed URL as the url attribute of a generic class-based view.
- providing a reversed URL to a decorator (such as the login_url argument for the django.contrib.auth.decorators.permission_required() decorator).
- providing a reversed URL as a default value for a parameter in a function’s signature.

Baschdl reviewed (approved):
Works now
